src/hls-audio-track.js

```diff
@@ -0,0 +1,88 @@
+import {AudioTrack} from 'video.js';
+import PlaylistLoader from './playlist-loader';
+
+/**
+ * HlsAudioTrack extends video.js audio tracks but adds HLS
+ * specific data storage such as playlist loaders, mediaGroups
+ * and default/autoselect
+ *
+ * @param {Object} options options to create HlsAudioTrack with
+ * @class HlsAudioTrack
+ * @extends AudioTrack
+ */
+export default class HlsAudioTrack extends AudioTrack {
+  constructor(options) {
+    super({
+      kind: options.default ? 'main' : 'alternative',
+      enabled: options.default || false,
+      language: options.language,
+      label: options.label
+    });
+
+    let trackProps = {
+      autoselect: options.autoselect || false,
+      default: options.default || false,
+      withCredentials: options.withCredentials || false
+    };
+
+    /* eslint-disable no-loop-func */
+    for (let key in trackProps) {
```

Comment:
I will implement the properties using Object.defineProperties as that sounds better.

They don't have to be custom properties, but I think that it makes it more consistent since they should never change, and since AudioTrack in video.js uses custom properties for everything already.